lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1930,6 +1930,29 @@ ParseResult FIRStmtParser::parsePrintf() {
     return failure();
 
   auto formatStrUnescaped = FIRToken::getStringValue(formatString);
+  StringRef formatStringRef(formatStrUnescaped);
+  if (formatStringRef.startswith("cover:")) {
+    APInt constOne(1, 1, false);
+    auto constTrue = builder.create<ConstantOp>(
+        info.getLoc(), UIntType::get(getContext(), 1), constOne);
+    builder.create<CoverOp>(info.getLoc(), clock, condition, constTrue,
+                            builder.getStringAttr(formatStrUnescaped));
+    return success();
+  } else if (formatStringRef.startswith("assert:")) {
+    APInt constOne(1, 1, false);
+    auto constTrue = builder.create<ConstantOp>(
+        info.getLoc(), UIntType::get(getContext(), 1), constOne);
+    builder.create<AssertOp>(info.getLoc(), clock, condition, constTrue,
+                             builder.getStringAttr(formatStrUnescaped));
+    return success();
+  } else if (formatStringRef.startswith("assume:")) {
```

Comment:
Mega-nit: since each condition is going to return, then you can drop the `else if` and just use `if`. Not blocking.

Reply:
Thanks, refactored.